broadinstitute/scp-ingest-pipeline
# Fix imprecise name sanitization in DE (SCP-4459)

Cluster name, annotation and annotation label sanitation for DE output files is currently collapsing down adjacent non-[^a-zA-Z0-9_] characters into one underscore. This can cause a file name mismatch which would prevent display of DE results.

Manual test:

activate the scp-ingest-pipeline repo virtualenv
then from the scripts directory of the scp-ingest-pipeline repo, perform this setup:

> source ../scripts/setup_mongo_dev.sh <path to your Github token file>
> unset BARD_HOST_URL

Run the DE job from the ingest directory of the scp-ingest-pipeline repo:

`python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 differential_expression --annotation-name misc++cellaneous --annotation-type group --annotation-scope study --matrix-file-path ../tests/data/differential_expression/de_dense_matrix.tsv --matrix-file-type dense --annotation-file ../tests/data/differential_expression/de_dense_metadata_sanitize.txt --cluster-file ../tests/data/differential_expression/de_dense_cluster.tsv --cluster-name "UMAP, pre-QC all cells (complexity greater than or equal to 1000)" --study-accession SCPsanitize --differential-expression`

confirm that the job runs successfully and the output files have more than one underscore where contiguous non-[^a-zA-Z0-9_] characters occur. (Note: the metadata file had non-[^a-zA-Z0-9_] characters in the annotation labels)

```
UMAP__pre_QC_all_cells__complexity_greater_than_or_equal_to_1000_--misc__cellaneous--cholinergic__neuron_--study--wilcoxon.tsv
UMAP__pre_QC_all_cells__complexity_greater_than_or_equal_to_1000_--misc__cellaneous--cranial__somatomotor__neuron--study--wilcoxon.tsv
UMAP__pre_QC_all_cells__complexity_greater_than_or_equal_to_1000_--misc__cellaneous--pyramidal__neuron--study--wilcoxon.tsv
UMAP__pre_QC_all_cells__complexity_greater_than_or_equal_to_1000_--misc__cellaneous--somatomotor_neuron_--study--wilcoxon.tsv
UMAP__pre_QC_all_cells__complexity_greater_than_or_equal_to_1000_--misc__cellaneous--sympathetic__cholinergic_neuron--study--wilcoxon.tsv
```

This PR satisfies SCP-4459.
